google/gax/utils/protobuf.py

```diff
@@ -124,7 +124,27 @@ def set(pb_or_dict, key, value):
     # with.
     if isinstance(pb_or_dict, collections.MutableMapping):
         pb_or_dict[key] = value
-    else:  # isinstance(pb_or_dict, Message):
+    elif isinstance(value, (collections.MutableSequence, tuple)):
+        # Clear the existing repeated protobuf message of any elements
+        # currently inside it.
+        while len(getattr(pb_or_dict, key)) > 0:
```

Comment:
I think it's [`pb_or_dict.ClearField(key)`](https://developers.google.com/protocol-buffers/docs/reference/python/google.protobuf.message.Message-class)?